### PR TITLE
Split colabfold workflow into two modes (colabfold_local and colabfold_webserver) and enable the usage of a custom webserver

### LINKED ISSUE #19: Add capability for localcolabfold to run vs a locally hosted database (FSx Lustre or local web-server)
### Description of feature

Currently, the localcolabfold.nf script is relying on a colabfold_batch command against the public web server hosted by the ColabFold developers (https://github.com/sokrypton/ColabFold) . This may not be obvious to first-time users of colabfold, because the workflow is checking for af2 native database files in the database path. The workflow does not use the database files if the mode is specified as colabfold. 

The issue with using the public web server is that it is a limited resource. After submitting anywhere from 10-100 jobs, additional jobs get killed by the web server (see attached file "command-2.err.txt" [command-2.err.txt](https://github.com/nf-core/proteinfold/files/8717522/command-2.err.txt)) and IP addresses can remain blocked for some time (1-2 days). This is necessary to keep the webserver operational and serving the community. 

A scalable solution for running colabfold would be highly beneficial to all nf-core/proteinfold users. Two options exist:

Solution 1: querying against a locally-installed colabfold db. The colabfold sequence dbs, which are distinct from the af2 native sequence dbs due to the fact colabfold uses mmseqs2 rather than jackhmmer for MSA generation, could be installed locally on an FSx Lustre filesystem. AWS developers have provided [such a solution using FSx](https://github.com/aws-samples/aws-batch-architecture-for-alphafold) . NOTE this would also required the Issue "Split colabfold_batch command in localcolabfold.nf into two separate, colabfold_search + colabfold_batch, commands" to be resolved, since colabfold_batch will not run against a file system, only a webserver. 

Solution 2: Provide an option for users to pass their own, locally hosted webserver url to the workflow. The [local webserver can be set up following instructions here](https://github.com/sokrypton/ColabFold/tree/main/MsaServer) . 

Of the two solutions, Solution 1 is preferable for most users. This is because it only requires that a database be maintained in a file system. Solution 2 would require a webserver to be running constantly, or to be configured and launched as part of the pipeline execution (overly complex). The webserver requires at least 2 TB of RAM, which amounts to a considerable overhead if running constantly.

### PR DESCRIPTION
**Pull request by athbaltzis**
## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [X] Usage Documentation in `docs/usage.md` is updated.
- [X] Output Documentation in `docs/output.md` is updated.
- [X] `README.md` is updated (including new tool citations and authors/contributors).

fixes #19 #20

### CODE REVIEW

**Review by JoseEspinosa — APPROVED** [review]
Nice @athbaltzis 🚀 
Made some suggestions, feel free to merge once addressed.
I have other comments that I will add on my own PR that will contain the rearrangement of the part of the code downloading the DBs.

File: conf/crg.config [hunk 1]
```diff
@@ -3,6 +3,10 @@ profiles {
     crg_cpu {
```
[the remark this answers is not in the document]
Comment (athbaltzis) [reply conf/crg.config:3]: At some point we have to create a general crg config file

File: conf/modules.config [hunk 2]
```diff
@@ -48,33 +46,52 @@ if (params.mode == 'AF2') {
...
 
-if (params.mode == 'colabfold') {
+if (params.mode == 'colabfold_local' | params.mode == 'colabfold_webserver') {
     process {
         withName: 'RUN_COLABFOLD' {
             ext.args = [
                 params.use_gpu ? '' : '--cpu',
-                params.use_amber ? '--amber' : ''
+                params.use_amber ? '--amber' : '',
+                params.host_url ? "--host-url ${params.host_url}" : ''
```
[the remark this answers is not in the document]
Comment (athbaltzis) [reply conf/modules.config:63]: Good point! No, it's redundant. I will update it

File: docs/output.md [hunk 3]
```diff
@@ -26,7 +27,7 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and predicts pr
 
 </details>
 
-### LocalColabFold
+### ColabFold
```
[the remark this answers is not in the document]
Comment (athbaltzis) [reply docs/output.md:30]: Sure! I will include it in the commit before merging

File: modules/local/mmseqs2.nf [hunk 4]
```diff
@@ -0,0 +1,30 @@
+process RUN_MMSEQS2 {
+    tag "${seq_name}"
+    label 'high'
```
[the remark this answers is not in the document]
Comment (athbaltzis) [reply modules/local/mmseqs2.nf:3]: No, this label (in the crg.conf) is for testing the `colabfold_local` mode which requires a lot memory. It is similar to the label `process_high_memory` in the base.conf and I plan to do this replacement in the next PR after finishing some tests.

**Pull request merged**